### PR TITLE
Resolve `npm audit` failure and make `jsdoc-to-markdown` "optional"

### PR DESCRIPTION
**Pull request by bsclifton**
(by introducing another dependency that will ONLY install if `npm run jsdoc` is used)

### CODE REVIEW

**Comment by bsclifton** [comment]
How this works:
- by default `jsdoc-to-markdown` (which has vulnerabilities) is not installed 👍 
- if you run `npm run jsdoc` (to update the docs), `jsdoc-to-markdown` will be fetched at that time
- after `npm run jsdoc` finishes, it unfortunately adds `jsdoc-to-markdown` to the package.json. So you're not gonna want to check that in after updating docs!

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "brave-crypto",
-  "version": "0.2.3",
+  "version": "0.2.4",
```
Comment (diracdeltas) [inline package.json]: do you need me to do a release to the npm registry? if not i'd prefer to keep the version number the same but no big deal either way

File: package.json [hunk 2]
```diff
@@ -45,6 +51,7 @@
   "dependencies": {
     "bip39": "^2.5.0",
     "niceware": "^1.0.5",
+    "optional-dev-dependency": "^2.0.1",
```
Comment (diracdeltas) [inline package.json]: does this work if this is in devDependencies instead of dependencies?

Comment (bsclifton) [reply package.json]: yes - it absolutely should 😄

File: package.json (hunk 1 above)
Comment (bsclifton) [reply package.json]: no update needed since `brave-core` is referencing the hashes via `DEPS` file 👍

File: package.json (hunk 2 above)
Comment (bsclifton) [reply package.json]: @diracdeltas moved! (refresh needed since I did a force push 😄)

File: package.json (hunk 1 above)
Comment (diracdeltas) [reply package.json]: @bsclifton could you please remove this line then? (keep the version at 0.2.3)

Comment (bsclifton) [reply package.json]: @diracdeltas done! 😄

**Pull request merged**